src/components/layout/Layout.jsx

```diff
@@ -8,73 +8,62 @@ import { AppContext } from '@edx/frontend-platform/react';
 import { getAuthenticatedUser } from '@edx/frontend-platform/auth';
 
 import EdXLogo from '../../assets/edx-logo.svg';
-
 import './styles/Layout.scss';
 
-class Layout extends Component {
-  getUserMenuItems = () => {
-    const { header: { userMenu } } = this.context;
-    return userMenu || [];
-  };
-
-  getMainMenuItems = () => {
-    const { header: { mainMenu } } = this.context;
-    return mainMenu || [];
-  };
+const Layout = ({
+  children,
+  headerLogo,
+  footerLogo,
+}) => {
+  const context = useContext(AppContext);
+  const { pageContext } = context;
 
-  render() {
-    const {
-      siteUrl, siteName, children, headerLogo, footerLogo,
-    } = this.props;
+  const user = getAuthenticatedUser();
+  const { username, profileImage } = user;
```

Comment:
I think we should make `profileImage = {}` as a default here since I occasionally see the error where profile image is undefined at the time layout is rendering

Reply:
> I think we should make `profileImage = {}` as a default here since I occasionally see the error where profile image is undefined at the time layout is rendering

Yeah, I did file a ticket for that issue (https://openedx.atlassian.net/browse/ENT-2652). I filed it as a ticket so we might take a deeper look into why we're rendering `Layout` when the hydrated user metadata isn't available.

Adding `profileImage = {}` will resolve that JS error, though so I will include that in this PR so we don't see that JS error anymore. Let's keep the above ticket open to revisit how to avoid rendering the app before all the authenticated user details are hydrated, though.